ReactAndroid/src/main/java/com/facebook/react/devsupport/InspectorPackagerConnection.java

```diff
@@ -195,39 +194,35 @@ public void onOpen(WebSocket webSocket, Response response) {
     }
 
     @Override
-    public void onFailure(IOException e, Response response) {
-      if (mWebSocket != null) {
-        abort("Websocket exception", e);
+    public void onFailure(WebSocket webSocket, Throwable t, Response response) {
+      if (webSocket != null) {
+        abort("Websocket exception", t);
       }
       if (!mClosed) {
         reconnect();
       }
     }
 
     @Override
-    public void onMessage(ResponseBody message) throws IOException {
+    public void onMessage(WebSocket webSocket, String text) {
       try {
-        handleProxyMessage(new JSONObject(message.string()));
+        handleProxyMessage(new JSONObject(text));
       } catch (JSONException e) {
         throw new IOException(e);
-      } finally {
-        message.close();
-      }
+      } 
     }
 
-    @Override
-    public void onPong(Buffer payload) {
-    }
 
     @Override
-    public void onClose(int code, String reason) {
+    public void onClosing(WebSocket webSocket, int code, String reason) {
+      webSocket.close(1000, null);
```

Comment:
Should I add some comments? 
1000 is normal closure status,
null is the reason (text)
https://github.com/square/okhttp/blob/master/samples/guide/src/main/java/okhttp3/recipes/WebSocketEcho.java